superset-frontend/src/explore/components/controls/AnnotationLayerControl/AnnotationLayer.jsx

```diff
@@ -131,128 +131,51 @@ const NotFoundContent = () => (
   </NotFoundContentWrapper>
 );
 
-class AnnotationLayer extends React.PureComponent {
-  constructor(props) {
-    super(props);
-    const {
-      name,
-      annotationType,
-      sourceType,
-      color,
-      opacity,
-      style,
-      width,
-      showMarkers,
-      hideLine,
-      value,
-      overrides,
-      show,
-      showLabel,
-      titleColumn,
-      descriptionColumns,
-      timeColumn,
-      intervalEndColumn,
-      vizType,
-    } = props;
-
-    // Only allow override whole time_range
-    if ('since' in overrides || 'until' in overrides) {
-      overrides.time_range = null;
-      delete overrides.since;
-      delete overrides.until;
-    }
-
-    // Check if annotationType is supported by this chart
-    const metadata = getChartMetadataRegistry().get(vizType);
-    const supportedAnnotationTypes = metadata?.supportedAnnotationTypes || [];
-    const validAnnotationType = supportedAnnotationTypes.includes(
-      annotationType,
-    )
-      ? annotationType
-      : supportedAnnotationTypes[0];
-
-    this.state = {
-      // base
-      name,
-      annotationType: validAnnotationType,
-      sourceType,
-      value,
-      overrides,
-      show,
-      showLabel,
-      // slice
-      titleColumn,
-      descriptionColumns,
-      timeColumn,
-      intervalEndColumn,
-      // display
-      color: color || AUTOMATIC_COLOR,
-      opacity,
-      style,
-      width,
-      showMarkers,
-      hideLine,
-      // refData
-      isNew: !name,
-      isLoadingOptions: true,
-      valueOptions: [],
-    };
```

Comment:
We need all of these values to go into the react state, but it looks like it only added the ones that had a key/value that were different.